### PR TITLE
chore: rename classes and use lombok when possible

### PR DESCRIPTION
**Pull request by KeranYang**
* Apply java google style. Most of the changes are automatic adjustments by IntelliJ.
* Use Lombok when possible.

Signed-off-by: Keran Yang <yangkr920208@gmail.com>

### CODE REVIEW

**Comment by vigith** [comment]
let's wait for #12 to be merged before we make this change?

**Comment by KeranYang** [comment]
> let's wait for https://github.com/numaproj/numaflow-java/pull/12 to be merged before we make this change?

Sure thing.

File: README.md [hunk 1]
```diff
@@ -57,3 +57,6 @@ option java_package = "io.numaproj.numaflow.function.v1";
 ```protobuf
 option java_package = "io.numaproj.numaflow.sink.v1";
 ```
+
+## Code Style
+If you use IntelliJ as Java IDE, use [google style guide](https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml).
```
Comment (yhl25) [inline README.md:62]: we support editorConfig. Do we need this?

Comment (KeranYang) [reply README.md:62]: Oh, I didn't know about editorConfig. Could you elaborate more on how to use it? Would like to ensure multiple developers follow the same coding style for this package.

Comment (yhl25) [reply README.md:62]: https://www.jetbrains.com/help/idea/editorconfig.html#:~:text=style%20to%20an%20.-,editorconfig%20file%EF%BB%BF,of%20the%20IDE%2Dlevel%20schemes.&text=%2C%20select%20Export%20and%20then%20EditorConfig%20File.

Comment (yhl25) [reply README.md:62]: editorConfig file is present in the project's root directory, you just need to point the ide to use that while formatting..

Comment (KeranYang) [reply README.md:62]: Thanks! Will do. +1

**Comment by KeranYang** [comment]
I messed up merging conflict  :(  and decided to move to a brand-new PR :)

@vigith , @yhl25 please review:  https://github.com/numaproj/numaflow-java/pull/16

**Pull request closed without merging**